### PR TITLE
Feature/import elementare gefaehrdungen

### PR DESCRIPTION
**Pull request by viktorianer**

### CODE REVIEW

**Comment by murygin** [comment]
Please attach a German IT Baseline Protection Catalog with basis threats.

**Comment by viktorianer** [comment]
[it-grundschutz_el14_eg_html_de.zip](https://github.com/SerNet/verinice/files/223696/it-grundschutz_el14_eg_html_de.zip)

File: sernet.gs.service/src/sernet/gs/scraper/GSScraper.java [hunk 1]
```diff
@@ -243,7 +243,14 @@ private void createBrokenRoleReplacements() {
 		} catch (XPathException e) {
 			Logger.getLogger(GSScraper.class).error(e);
 			throw new GSServiceException(e);
-		}
+        } catch (GSServiceException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(e.getMessage());
```
Comment (murygin) [inline sernet.gs.service/src/sernet/gs/scraper/GSScraper.java]: Please log message and stack trace.

File: sernet.gs.service/src/sernet/gs/scraper/GSScraper.java [hunk 2]
```diff
@@ -243,7 +243,14 @@ private void createBrokenRoleReplacements() {
 		} catch (XPathException e) {
 			Logger.getLogger(GSScraper.class).error(e);
 			throw new GSServiceException(e);
-		}
+        } catch (GSServiceException e) {
+            if (LOG.isDebugEnabled()) {
```
Comment (murygin) [inline sernet.gs.service/src/sernet/gs/scraper/GSScraper.java]: Please log message and stack trace. Moreover use log level _error_ if _kapitel_ is not equals _"b00"_. Use a constant for string _"b00"_.

File: sernet.gs.service/src/sernet/gs/scraper/GSScraper.java [hunk 3]
```diff
@@ -366,7 +373,14 @@ private void getStand(String kapitel, Node root) throws XPathException {
...
```
Comment (murygin) [inline sernet.gs.service/src/sernet/gs/scraper/GSScraper.java]: Please log message and stack trace. Moreover use log level error if kapitel is not equals "b00000". Use a constant for string "b00000".

File: sernet.gs.service/src/sernet/gs/scraper/ZIPGSSource.java [hunk 4]
```diff
@@ -183,7 +183,7 @@ public Node parseBausteinDocument(String bausteinFileName) throws GSServiceExcep
 			
 		
 		} catch (Exception e) {
```
Comment (murygin) [inline sernet.gs.service/src/sernet/gs/scraper/ZIPGSSource.java]: Why do you switch from _error_ to _warn_ here and why do you not log the stack trace anymore? Please use the _Logger LOG_ defined in the class.

File: sernet.gs.ui.rcp.main/src/sernet/gs/ui/rcp/main/bsi/views/BSIMassnahmenView.java [hunk 5]
```diff
@@ -363,12 +363,13 @@ public boolean hasChildren(Object parent) {
...
             // as Sonar suggests breaks the tree in the IT Baseline Protection Catalog View.
-            if (parent instanceof Baustein) {
-                return ((Baustein) parent).getMassnahmen().size() > 0;
-            } else if (parent instanceof BSIKatalogInvisibleRoot) {
-                return ((BSIKatalogInvisibleRoot) parent).getBausteine().size() > 0;
-            }
-            return false;
+            if (parent instanceof Baustein){
+                Baustein baustein = (Baustein) parent;
+				return baustein.getMassnahmen().size() > 0 || baustein.getGefaehrdungen() != null ? baustein.getGefaehrdungen().size() > 0 : false;
```
Comment (murygin) [inline sernet.gs.ui.rcp.main/src/sernet/gs/ui/rcp/main/bsi/views/BSIMassnahmenView.java]: This statement is confusing and hard to understand. You can make this more readable by introducing a new method.

**Comment by murygin** [comment]
Please write a description for this pull request including a summary of the feature request.

**Comment by murygin** [comment]
Code review finished. See my comments and change code where required.

**Pull request merged**